src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonMetadataServices.Helpers.cs

```diff
@@ -124,7 +124,7 @@ private static void PopulateProperties(JsonTypeInfo typeInfo, Func<JsonSerialize
             Debug.Assert(typeInfo.Kind is JsonTypeInfoKind.Object);
             Debug.Assert(!typeInfo.IsReadOnly);
 
-            JsonSerializerContext? context = typeInfo.Options.TypeInfoResolver as JsonSerializerContext;
+            JsonSerializerContext? context = (typeInfo.OriginatingResolver ?? typeInfo.Options.TypeInfoResolver) as JsonSerializerContext;
```

Comment:
when can typeInfo.OriginatingResolver be null?

Reply:
Older versions of the source generator.